neo/rawio/nixrawio.py

```diff
@@ -359,23 +377,20 @@ def _rescale_epoch_duration(self, raw_duration, dtype='float64'):
         # supposing unit is second, other possibilities maybe mS microS...
         return durations  # return in seconds
 
-    def _add_annotate(self, tar_ann, props, otype):
-        values = props.values
-        list_of_param = []
-        if otype == 'seg':
-            list_of_param = ['index']
-        elif otype == 'asig':
-            list_of_param = ['units', 'copy', 'sampling_rate', 't_start']
-        elif otype == 'st':
-            list_of_param = ['units', 'copy', 'sampling_rate',
-                             't_start', 't_stop', 'waveforms', 'left_sweep']
-        elif otype == 'ev':
-            list_of_param = ['times', 'labels', 'units', 'durations', 'copy']
-        if len(values) == 1:
-            values = values[0]
-        if props.name not in list_of_param:
-            tar_ann[str(props.name)] = values
-        else:
-            warntxt = "Name of annotation {} shadows parameter " \
-                        "and is therefore dropped".format(props.name)
-            warnings.warn(warntxt)
+    def _filter_properties(self, properties, neo_type):
+        """
+        Takes a collection of NIX metadata properties and the name of a Neo
+        type and returns a dictionary representing the Neo object annotations.
+        Properties that represent the attributes of the Neo object type are
+        filtered, based on the global 'neo_attributes' dictionary.
+        """
+        annotations = dict()
+        # ignore neo_name as well
+        attrs = neo_attributes.get(neo_type, list()) + ["neo_name"]
+        for prop in properties:
+            if not (prop.name in attrs or prop.name == "neo_name"):
```

Comment:
isn't this handling `neo_name` twice? once in line 389 and once in line 391?

Reply:
You're right.  Let me clean that up.